Api/Src/Services/GradeSheetService.ts

```diff
@@ -46,15 +59,75 @@ export default class GradeSheetService {
     }
 
     async setMentorReviewers(gradeSheetId: mongoose.Types.ObjectId, mentorIds: mongoose.Types.ObjectId[]) {
-        return await this.repository.setMentorReviewers(gradeSheetId, mentorIds);
+        const sheet = await this.repository.getById(gradeSheetId) as GradeSheet & mongoose.Document | null;
+        sheet.reviewers = sheet.reviewers
+            .filter(rev => mentorIds.some(id => `${rev}` === `${id}`));
+        sheet.mentorReviewerGrades = sheet.mentorReviewerGrades
+            .filter(grade => mentorIds.some(id => `${grade.mentorID}` === `${id}`));
+        for (let i in mentorIds) {
+            if (sheet.reviewers.some(rev => `${mentorIds[i]}` === `${rev}`)) continue;
+            sheet.reviewers.push(mentorIds[i]);
+            sheet.mentorReviewerGrades.push({mentorID: mentorIds[i], grades: {}});
+        }
+        return await this.repository.save(sheet);
```

Comment:
it is possible to set mentor reviewer to non-existing user id
I think we can live with that though as this is manageable on frontend